tests/RuleFilterParser.Tests/LinqTests.cs

```diff
@@ -0,0 +1,339 @@
+﻿using RuleFilterParser;
+using RuleToLinqParser.Tests.Helpers;
+using RuleToLinqParser.Tests.Seeds;
+
+namespace RuleToLinqParser.Tests;
+
+public class LinqQueryProviderTests
+{
+    private ApiClientMockup InitializeApiClient()
+    {
+        var api = new ApiClientMockup();
+        api.RegisterData(typeof(Person), Persons.TestData());
+        return api;
+    }
+
+    [Fact]
+    public void SelectTest()
+    {
+        var api = InitializeApiClient();
+        var personsSource = api.GetAsQueryable<Person>("person");
```

Comment:
Isn't `person` just an entity name? Wouldn't it be better to use strongly typing like this?

` await CoreApi.Person.FindAsync();`

Reply:
I think we need async overload as well. Or preferable only async as we will use it in async communication.

Reply:
That is true, the direct call of `GetAsQueryable<Person>(...)` is just cumbersome, I can change that. After all, this was just a mockup so I did not spend so much time at this. I was expecting that this would be implemented inside the real API client. I focused mostly on the linq expression processing engine.

Reply:
Now the mockup API client implements strongly typing as well.

Reply:
> I think we need async overload as well. Or preferable only async as we will use it in async communication.

The Queryable is never Async since it is only creating the query, not executing it. The execution happens when you call something like .ToListAsync() on the queryable (at least in the Core API implementation).

If we are to support both sync and async it's important to be consistent all the way down to the http request (and not use things like .Result to make things sync).

Reply:
I suggest that we stick to the Async all the way down.